api/auth/auth.controller.js

```diff
@@ -0,0 +1,88 @@
+const {
+  Types: { ObjectId },
+} = require("mongoose");
+
+const User = require("../users/users.model");
+const bcrypt = require("bcrypt");
+const { createVerificationToken } = require("../../services/token.service");
+const { avatar } = require("../../avatarCreator");
+
+const registrationController = async (req, res, next) => {
+  try {
+    const { password, email, subscription } = req.body;
+    const user = await User.findUser({ email });
+    if (user) {
+      return res.status(409).send("Email in use");
+    }
+    const hashedPassword = await bcrypt.hash(
+      password,
+      Number(process.env.SALT)
+    );
+    const generateAvatarId = Date.now();
+    const variant = "female";
+    const image = await avatar.generate("email@example.com", variant);
+    const savedAvatar = await image
+      .png()
+      .toFile(`./public/images/${generateAvatarId}.png`);
```

Comment:
move `./public/images/${generateAvatarId}.png` to config 

const yourFnName = (id) => `./public/images/${id}.png`